library/src/androidTest/java/com/raizlabs/android/dbflow/test/sql/BoxedValueTest.java

```diff
@@ -3,24 +3,129 @@
 import android.database.sqlite.SQLiteConstraintException;
 import android.database.sqlite.SQLiteException;
 
+import com.raizlabs.android.dbflow.sql.builder.Condition;
+import com.raizlabs.android.dbflow.sql.language.Select;
 import com.raizlabs.android.dbflow.test.FlowTestCase;
 
+import java.util.concurrent.atomic.AtomicLong;
+
 /**
  * Description: Test to ensure that passing null to non-null fields does not cause a NPE and that it
  * will fail.
  */
 public class BoxedValueTest extends FlowTestCase {
 
-    public void testBoxedValues() {
-        BoxedModel boxedModel = new BoxedModel();
-        boxedModel.id = null;
-        boxedModel.rblNumber = null;
+    private static final AtomicLong SEQUENCE_ID = new AtomicLong();
+    private BoxedModel testObject;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        testObject = new BoxedModel();
+        testObject.id = SEQUENCE_ID.getAndIncrement();
+    }
+
+    public void testBoxedValues_nullId() {
```

Comment:
dont believe we need to test every field here to see if its saveable or not. rather all we need to test is a non-null and normal field can be saved.